src/event_handlers.py

```diff
@@ -0,0 +1,140 @@
+# pylint: disable=too-few-public-methods
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Union
+
+from celery import Task
+from loguru import logger
+
+from src.constants import EventEnum, EventType, LabelName
+from src.helpers import get_exception_class
+from src.instrumentation import EventCounter, EventGauge
+
+
+class InvalidTaskEventLabelName(Exception):
+    pass
+
+
+class IEventHandler(ABC):
+    def __init__(self, state):
```

Comment:
Unfortunately pylint does not see
`from celery.events.state import State` as a valid import (despite Python itself does!) and will raise:
`src/event_handlers.py:7:0: E0611: No name 'state' in module 'celery.events' (no-name-in-module)`
So I skipped typing the `state` for now cause setting an ignore for the whole file on `no-name-in-module` felt bad.